### PR TITLE
[PRFC] Previewing Scaffolder Template UI Forms

### PR DESCRIPTION
**Pull request by kuangp**
Signed-off-by: Phil Kuang <pkuang@factset.com>

Not sure if this qualifies as an RFC but putting up something I hacked together to help preview and debug template form UIs. I have this setup as a separate route (`/create/preview`) taking inspiration from `/create/actions` but both seem fairly hidden right now which I think both should be linked from the main `ScaffolderPage` somehow to make them more discoverable, open to ideas here. Also open to any suggestions for better naming for stuff. 

https://user-images.githubusercontent.com/6998196/155904837-f6a24344-5c51-4ca4-a689-4c7c7f175ee6.mov

## Need

An easy way to preview scaffolder template form UIs without running your own local instance of the plugin or committing changes to the template.  

## Proposal

A view implemented directly in the scaffolder plugin reusing the same rendering component that powers the UI form to preview templates with live reload support for adhoc template drafting. This will allow us to also test and preview any custom field extensions installed in your Backstage instance. It addition, this also provides a "playground" that is somewhat agnostic to whatever rendering engine we use to power the scaffolder (eg. if we decide to swap out rjsf in the future, users can still use this previewer instead of switching to a new tool to preview their UIs ... and even more so if we abstract the template syntax) 

Some additional features I can see added in the future includes:
- More advanced editor features like autocomplete to insert example syntax for available custom field extensions
- Support for editing and validating the entire template entity spec (not just `parameters`) and support for inserting example syntax for available installed actions
-  Support for taking your draft changes and creating PRs for SCM backed existing templates or new entity files for creating a new template from your draft
- Linking/embedding docs from `ActionsPage` along with docs for available custom fields to use as dev references

## Alternatives
https://rjsf-team.github.io/react-jsonschema-form/

The above playground is currently the recommended option in the [docs](https://backstage.io/docs/features/software-templates/writing-templates#specparameters---formstep--formstep) to test how your form will look in the UI but I found this unintuitive since it expects JSON while templates are written in YAML. It also doesn't support testing any custom field extensions and not every feature seems to be directly supported/rendered the same in the scaffolder which can lead to confusion.

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by regicsolutions** [comment]
This looks awesome, will save tons of time!

**Comment by satrox28** [comment]
Looks great, it will be a good addition to Backstage 👍

**Comment by peimanja** [comment]
This is amazing !

**Comment by freben** [comment]
Wow! Just wanted to say we've been extra swamped the last couple of days but we'll get to this eventually!

**Comment by benjdlambert** [comment]
@kuangp looks like some conflicts crept in here, could you fix it up and we can review :pray:

**Review by benjdlambert — COMMENTED** [review]
Thanks @kuangp! :tada: just one small note.

File: packages/app/src/App.tsx [hunk 1]
```diff
@@ -88,6 +88,45 @@ import { ApacheAirflowPage } from '@backstage/plugin-apache-airflow';
 import { PermissionedRoute } from '@backstage/plugin-permission-react';
 import { catalogEntityCreatePermission } from '@backstage/plugin-catalog-common';
 
+const defaultPreviewTemplate = `# Edit the template parameters below to see how they will render in the scaffolder form UI
```
[the remark this answers is not in the document]
Comment (kuangp) [reply packages/app/src/App.tsx]: makes sense! done!

**Review by benjdlambert — APPROVED** [review]
Nice work! Thanks!

**Review by benjdlambert — APPROVED** [review]

**Pull request merged**